azure-devops/enforce-clang-format.cmd

```diff
@@ -1,6 +1,5 @@
 :: Copyright (c) Microsoft Corporation.
 :: SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-@echo off
-clang-format -style=file -i stl/inc/* stl/inc/cvt/* stl/inc/experimental/* stl/src/* 2>&1
+"%1" "clang-format.exe -style=file -i" stl/inc stl/inc/cvt stl/inc/experimental stl/src tools/inc tools/jobify/jobify.cpp tools/parallelize/parallelize.cpp
 echo If your build fails here, you need to format the following files with clang-format 8.0.1.

```

Comment:
I think I prefer encoding the version that the repo expects into the repo itself is the right approach here; otherwise it is confusing which version was intended if you sync to an old commit and fail this test.

Reply:
But if there's mismatch between the version of clang-format that was used on an old commit, and the version that's being run in Azure Pipelines, then the PR is doomed anyways. I don't understand the concern here.

Reply:
If you run a build against an old PR, and get failures from some new version of clang format, and the message is talking about an old version of clang-format, it's easier to tell what happened than if whatever the current version of clang-format is is printed instead.

Reply:
I am still confused. How can the user know what version of clang-format was run without `clang-format --version` being printed? And if so, what good does it do for a separate version to be mentioned here? I could understand it if the *repo sources* had banners saying "formatted with clang-format 8.0.1" but they don't.

It is also unclear to me how someone could easily "run a build against an old PR". If they submit a PR, they will typically be building against the latest commit, right?